File: my_project_typescript/easy_problems_typescript/from151to200/validParentheses.ts
function isValid(s: string): boolean {

    let temp: string[] = []
    let validParentheses: string[] = ['(','[','{']

    for (let p of s){
        if (validParentheses.includes(p)){
            temp.push(p)
        }
        else{
            if (!temp){
                return false 
            }
            let other_half = temp.pop()
            if (p == ')'){
                if (other_half != '('){
                    return false;
                }
            }
            else if (p == ']'){
                if (other_half != '['){
                    return false 
                }
            }
            else if (p == '}'){
                if (other_half != '{'){
                    return false 
                }
            }
            else{
                return false 
            }

        }
    }


    if (temp.length>0){
        return false 
    }

    return true 

};

console.log(isValid('()'))

File: my_java_projects/src/easy_problems/from_1_to_50/ValidParentheses.java
package easy_problems.from_1_to_50;

import java.util.Stack;

public class ValidParentheses implements Numbers{
    public ValidParentheses() {
        System.out.println("An instance of the ValidParentheses class was called.");
    }

    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<Character>();
        for (char c : s.toCharArray()) {
            if (c == '(')
                stack.push(')');
            else if (c == '{')
                stack.push('}');
            else if (c == '[')
                stack.push(']');
            else if (stack.isEmpty() || stack.pop() != c)
                return false;
        }
        return stack.isEmpty();
    }

    @Override
    public void returnNumbers() {
        System.out.println("The instance of the class ValidParentheses do not return numbers");
    }
}
